test_tf2/test/test_message_filter.cpp

```diff
@@ -225,121 +252,133 @@ TEST(MessageFilter, multipleTargetFrames)
   filter.add(msg);
 
-  EXPECT_EQ(0, n.count_);
-
-  queue.callAvailable();
-
-  EXPECT_EQ(1, n.count_);
+  EXPECT_EQ(1, n.count_); // Latest message is off the end of the offset
+  rclcpp::shutdown();
 }
 
-
+// TODO(ahcorde): failurecallback
+// TEST(MessageFilter, outTheBackFailure)
+// {
+//   BufferCore bc;
+//   Notification n(1);
+//   MessageFilter<geometry_msgs::msg::PointStamped> filter(bc, "frame1", 1, 0);
+//   filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
+//
+//   builtin_interfaces::msg::Time stamp(1);
+//   bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp, "frame1", "frame2"), "me");
+//   bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp + tf2::durationFromSec(10000), "frame1", "frame2"), "me");
+//
+//   geometry_msgs::msg::PointStampedPtr msg(new geometry_msgs::msg::PointStamped);
+//   msg->header.stamp = stamp;
+//   msg->header.frame_id = "frame2";
+//   filter.add(msg);
+//
+//   EXPECT_EQ(1, n.failure_count_);
+// }
+//
+// TEST(MessageFilter, outTheBackFailure2)
+// {
+//   BufferCore bc;
+//   Notification n(1);
+//   MessageFilter<geometry_msgs::msg::PointStamped> filter(bc, "frame1", 1, 0);
+//   filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
+//
+//   builtin_interfaces::msg::Time stamp(1);
+//
+//   geometry_msgs::msg::PointStampedPtr msg(new geometry_msgs::msg::PointStamped);
+//   msg->header.stamp = stamp;
+//   msg->header.frame_id = "frame2";
+//   filter.add(msg);
+//
+//   EXPECT_EQ(0, n.count_);
+//   EXPECT_EQ(0, n.failure_count_);
+//
+//   bc.setTransform(createTransform(Quaternion(0,0,0,1), Vector3(1,2,3), stamp + tf2::durationFromSec(10000), "frame1", "frame2"), "me");
+//
+//   EXPECT_EQ(1, n.failure_count_);
+// }
+//
+// TEST(MessageFilter, emptyFrameIDFailure)
+// {
+//   BufferCore bc;
+//   Notification n(1);
+//   MessageFilter<geometry_msgs::msg::PointStamped> filter(bc, "frame1", 1, 0);
+//   filter.registerFailureCallback(boost::bind(&Notification::failure, &n, _1, _2));
+//
+//   geometry_msgs::msg::PointStampedPtr msg(new geometry_msgs::msg::PointStamped);
+//   msg->header.frame_id = "";
+//   filter.add(msg);
+//
+//   EXPECT_EQ(1, n.failure_count_);
+// }
+//
+// TEST(MessageFilter, callbackQueue)
+// {
+//   BufferCore bc;
+//   Notification n(1);
+//   ros::CallbackQueue queue;
+//   MessageFilter<geometry_msgs::msg::PointStamped> filter(bc, "frame1", 1, &queue);
+//   filter.registerCallback(boost::bind(&Notification::notify, &n, _1));
+//
+//   geometry_msgs::msg::PointStampedPtr msg(new geometry_msgs::msg::PointStamped);
+//   msg->header.stamp = builtin_interfaces::msg::Time(1);
+//   msg->header.frame_id = "frame1";
+//   filter.add(msg);
+//
+//   EXPECT_EQ(0, n.count_);
+//
+//   queue.callAvailable();
+//
+//   EXPECT_EQ(1, n.count_);
+// }
+//
+//
 int main(int argc, char** argv)
 {
   testing::InitGoogleTest(&argc, argv);
-
+  rclcpp::init(argc, argv);
```

Comment:
So you call `rclcpp::init` here, but you seem to call `rclcpp::shutdown` in several of the tests themselves.  Does that work?  I would think that would actually lead to problems, but I'm not sure.

Reply:
fixed